keps/sig-node/127-user-namespaces/README.md

```diff
@@ -0,0 +1,437 @@
+a trivial next-step to take, given that we have phase 1 implemented: just return
+the same mapping if the pod has other volumes.
+
+#### Phase 3: pod to pod isolation
+
+This phase will provide more isolation between pods that use volumes (as in
+phase 2) and requires another opt-in field:
+`pod.spec.securityContext.pod2podIsolation`.
+
+This phase will try to not share the same mapping for all pods with volumes, as
+phase 2 does, but to achieve it some trade off needs to be made.
+
+One idea is to give different mappings to pods in different k8s namespaces or
+that use a different service account. This needs to be explored in further
+detail, but will probably impose limits to which workloads can run this (we need
+to expose a shorter mapping, less than 64k).
+
+Another idea is to use id mapped mounts. This probably needs changes to the
+OCI runtime-spec, only works with certain filesystems and kernels that may take
+too long for some users to get (like managed services). Giuseppe started to
+experiment in crun with this
+[here](https://github.com/containers/crun/pull/780).
+
+This field value will be an enum, to select different strategies and allow room for
+future improvements.
+
+It is being considered having a value that is "auto" for this fields, that
+will select the best strategy that your node supports. However, as different
+strategies will change the effective UID a container uses, if we add such an
+option the documentation will be VERY clear about the implications and
+automatizations will be provided whenever possible (we have some ideas on this
+front).
+
+This phase builds on the work of the previous phases and more details will be
+defined while the other phases evolve.
+
+If this phase turns out to be a lot of work, it will be left out as future work
+for other KEPs.
+
+### Summary of the Proposed Changes
+
+[This table](https://docs.google.com/presentation/d/1z4oiZ7v4DjWpZQI2kbFbI8Q6botFaA07KJYaKA-vZpg/edit#slide=id.gfd10976c8b_1_41) gives you a quick overview of each phase.
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+TBD
+
+### Graduation Criteria
+
+Graduation for each pod.spec field we introduce will be separate.
```

Comment:
I want to highlight this. I'd like the different pod.spec changes to graduate separate, so phase 1&2 can be in GA but phase 3 still in alpha/beta, as we gather more info and feedback (phase 3 will probably need quite a lot of feedback, I don't want that to block the progress of the rest of the features)

Reply:
That will be through separate feature flags, right? Do we have proposed names? If we don't have one for later phases, we should call it out as a TBD.